### ISSUE TITLE
[Wine] workspace rendering issue

### ISSUE THREAD

**Issue by brickviking** [dolphinsmalltalk/Dolphin#693]
**TL;DR**
 If I don't have any Workspaces open when I start up Dolphin, the System Folder and other windows appear correctly. If I have a workspace shown, sometimes other windows won't render either.

**Details**
At least under Wine, Workspaces (and sometimes other windows if Workspaces are shown) are initially unexposed, and their contents aren't showing. I saved an image from Vista (looking perfectly normal) and copied it over to Linux. I started Dolphin up under wine, and I got the following picture to try to illustrate the problem I have, showing two workspaces and the System Folder:

![dolphindesktop-shrunk](https://user-images.githubusercontent.com/62885/49640316-2a700400-fa72-11e8-9948-6121493b988c.png)

Additionally, the three workspace toolbars on the right hand side aren't vertical, they're all horizontal. I can't get the toolbars to reset their alignment even by prodding values from an Inspector. Also, the three toolbars (doIt/evaluateIt..., find, cut/copy/paste) don't have an arrangement field, though it appears that a section of apparently empty toolbar above them has an arrangement of #north. Is there a layout or flow manager involved there somewhere? If so, I couldn't find it in the classes I tried looking at to see how a Workspace was built. (Man, that's confusing).

If I click on the titlebar of the Workspace, the contents do refresh, but only when I've selected the window. Editing the text inside the window works, but behaviour is ambiguous between some workspaces. One Workspaces' text can be edited, but I don't see the text being edited until (once again) I click on the titlebar. The other edits perfectly fine (i.e. I see the characters I'm typing in). I don't know if this is a wine error or something funky with the workspaces I've created. If I close all workspaces and create a new one, once again the main text widget doesn't show until I click on the titlebar, and I can't visibly edit the text.

I don't see the same issues with Class/Package/System Browsers; they all appear correctly and can be visibly interacted with. All their contents are visible and remain so until I open Dolphin again with a Workspace showing, after which all bets are off. Some windows render, some don't.

Any ideas about where to take this further?

Regards, BrickViking (a.k.a. DrSmokey)

**Comment by blairmcg** [comment]
The only thing I can suggest would be to try removing the vertical toolbar from the workspace view. It wouldn't particularly surprise me if that and a lot of other things didn't work correctly on Wine.

**Comment by brickviking** [comment]
Yeah, so it would seem. In the end, what I did was to shift the three sub-toolbars in the vertical container (toolbar2) up to the horizontal toolbar container (toolbar1), via using the Toolbar that contains the single Save (file) icon to flip the sub-toolbar around from vertical to horizontal. Wonky, but it worked. I haven't been able to figure out another way to flip the sub-toolbar around, and frankly I think this is a "workaround" for a UI issue. Anyhow, shifting the toolbars around allowed me to be able to consistently view Workspaces without any apparent issues, but getting to those resources meant evaluating a ViewComposer expression from within the comment of the class method SmalltalkWorkspaceDocument>>resource_Default_view

I had to of course do the editing over on Vista, because there's really something funky wrong under Wine with the rendering of the various browser windows in the ViewComposer. I haven't been able to put my finger on why; it could be Wine's "lot of other things" getting in the way.

Regards, BrickViking (a.k.a. DrSmokey)

**Comment by brickviking** [comment]
Further to the previous post, I basically nuked my copy of wine's drive, and got wine to create one for me. Once I added gdiplus, mfc40/42, comctl32 (not sure if that was needed) and vcrun2015, Dolphin workspace windows then rendered their text area properly. It's also apparent I no longer need a a ucrtbase.dll local to Dolphin7.exe. The icons arrangement in the right hand side toolbars are now correct—they're vertical, though the two toolbars appear truncated. However, some icons in the right toolbars don't render. I'll post another bug report regarding the inability to use the Search facility if I can nail down why I can't get it to show; I haven't had the search facility showing up for…well, a few months now. There's a spot in the upper toolbar where it plainly looks like a search bar would fit, but there's no text box there, just the toolbar background showing.

Is there a likely Z-order problem here? Say, toolbar background gets mapped in front of the textbox, so it doesn't show up and can't be selected. It's just a mad idea of mine.

**Comment by blairmcg** [comment]
Closing this, since as you say the workspace rendering issue to which it relates was fixed by rebuilding your Wine setup.
